module/Kanbanize/src/Kanbanize/KanbanizeTask.php

```diff
@@ -2,65 +2,115 @@
+use TaskManagement\Stream;
 
 class KanbanizeTask extends Task {
 	
 	//FIXME how to map backlog?
-	CONST BACKLOG = 'Backlog';
-	CONST COLUMN_IDEA = 'Idea';
-	CONST COLUMN_OPEN = 'Open';
-	CONST COLUMN_ONGOING = "OnGoing";
-	CONST COLUMN_COMPLETED = 'Completed';
-	CONST COLUMN_ACCEPTED = 'Accepted';
-
-	private static $mapping = array(
-			self::COLUMN_IDEA		=> Task::STATUS_IDEA,
-			self::COLUMN_OPEN		=> Task::STATUS_OPEN,
-			self::COLUMN_ONGOING	=> Task::STATUS_ONGOING,
-			self::COLUMN_COMPLETED	=> Task::STATUS_COMPLETED,
-			self::COLUMN_ACCEPTED	=> Task::STATUS_ACCEPTED,
-			Task::STATUS_IDEA		=> self::COLUMN_IDEA, 
-			Task::STATUS_OPEN		=> self::COLUMN_OPEN,
-			Task::STATUS_ONGOING	=> self::COLUMN_ONGOING,
-			Task::STATUS_COMPLETED	=> self::COLUMN_COMPLETED,
-			Task::STATUS_ACCEPTED	=> self::COLUMN_ACCEPTED,
-			//FIXME backlog?
-			self::BACKLOG			=> -1,
-			-1						=> self::BACKLOG
-	);
-	
-	private $kanbanizeBoardId;
-	
-	private $kanbanizeTaskId;
+// 	CONST BACKLOG = 'Backlog';
+// 	CONST COLUMN_IDEA = 'Idea';
+// 	CONST COLUMN_OPEN = 'Open';
+// 	CONST COLUMN_ONGOING = "OnGoing";
+// 	CONST COLUMN_COMPLETED = 'Completed';
+// 	CONST COLUMN_ACCEPTED = 'Accepted';
+// 	CONST COLUMN_CLOSED = 'Closed';
+
+	CONST EMPTY_ASSIGNEE = 'None';
+
+	//Mapping fo kanbanize columns is not static: is setted for each organization
+// 	private static $mapping = array(
+// 			self::COLUMN_IDEA		=> Task::STATUS_IDEA,
+// 			self::COLUMN_OPEN		=> Task::STATUS_OPEN,
+// 			self::COLUMN_ONGOING	=> Task::STATUS_ONGOING,
+// 			self::COLUMN_COMPLETED	=> Task::STATUS_COMPLETED,
+// 			self::COLUMN_ACCEPTED	=> Task::STATUS_ACCEPTED,
+// 			self::COLUMN_CLOSED		=> Task::STATUS_CLOSED,
+// 			Task::STATUS_IDEA		=> self::COLUMN_IDEA, 
+// 			Task::STATUS_OPEN		=> self::COLUMN_OPEN,
+// 			Task::STATUS_ONGOING	=> self::COLUMN_ONGOING,
+// 			Task::STATUS_COMPLETED	=> self::COLUMN_COMPLETED,
+// 			Task::STATUS_ACCEPTED	=> self::COLUMN_ACCEPTED,
+// 			Task::STATUS_CLOSED		=> self::COLUMN_CLOSED,
+// 			//FIXME backlog?
+// 			self::BACKLOG			=> -1,
+// 			-1						=> self::BACKLOG
+// 	);
+
+	/**
+	 * @var String
+	 */
+	private $taskid;
+	/**
+	 * @var String
+	 */
+	private $assignee;
+	/**
+	 * @var String
+	 */
+	private $columnname;
 	
 	public static function create(Stream $stream, $subject, BasicUser $createdBy, array $options = null) {
-		if(!isset($options['kanbanizeBoardId']) || !isset($options['kanbanizeTaskId'])) {
-			throw InvalidArgumentException('Cannot create a KanbanizeTask without a kanbanizeBoardId or kanbanizeTaskId option');
+		if(!isset($options['taskid'])) {
+			throw InvalidArgumentException('Cannot create a KanbanizeTask without a taskid option');
+		}
+		if(!isset($options['columnname'])) {
+			throw InvalidArgumentException('Cannot create a KanbanizeTask without a columnname option');
 		}
 		$rv = new self();
 		$rv->id = Uuid::uuid4();
-		$rv->status = self::STATUS_ONGOING;
+		$rv->status = $options["status"];
```

Comment:
Non ho voluto tenere un vincolo stretto sullo stato del task; immaginando ad esempio la prima sincronizzazione di una board con O.R.A. : se in quella board non ci fossero task in uno stato "work item idea" non importerei neanche un task. E' più corretto che sia così ?